From 9c657d7f8301265d62ef0c9dab5970437ffdd93e Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Tue, 23 May 2017 19:46:06 +0800
Subject: [PATCH 1842/5242] MLK-14968-2 ARM64: dts: fsl-imx8: add usdhc1
 support HS400 mode

commit  ab88011d508a373dd43f37d25b94b9d3ce41cf40 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add usdhc1 support for HS200/HS400 mode for imx8qm and imx8qxp.

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts  |   43 ++++++++++++++++++--
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi      |    5 ++-
 .../boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts |   43 ++++++++++++++++++--
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi     |    5 ++-
 4 files changed, 88 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
index 67120c9..b8fe58f 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
@@ -130,7 +130,7 @@
 
 		pinctrl_usdhc1: usdhc1grp {
 			fsl,pins = <
-				SC_P_EMMC0_CLK_CONN_EMMC0_CLK		0x06000021
+				SC_P_EMMC0_CLK_CONN_EMMC0_CLK		0x06000041
 				SC_P_EMMC0_CMD_CONN_EMMC0_CMD		0x00000021
 				SC_P_EMMC0_DATA0_CONN_EMMC0_DATA0	0x00000021
 				SC_P_EMMC0_DATA1_CONN_EMMC0_DATA1	0x00000021
@@ -140,13 +140,48 @@
 				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000021
 				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000021
 				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000021
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000041
+				SC_P_EMMC0_RESET_B_CONN_EMMC0_RESET_B	0x00000021
+			>;
+		};
+
+		pinctrl_usdhc1_100mhz: usdhc1grp100mhz {
+			fsl,pins = <
+				SC_P_EMMC0_CLK_CONN_EMMC0_CLK		0x06000045
+				SC_P_EMMC0_CMD_CONN_EMMC0_CMD		0x00000025
+				SC_P_EMMC0_DATA0_CONN_EMMC0_DATA0	0x00000025
+				SC_P_EMMC0_DATA1_CONN_EMMC0_DATA1	0x00000025
+				SC_P_EMMC0_DATA2_CONN_EMMC0_DATA2	0x00000025
+				SC_P_EMMC0_DATA3_CONN_EMMC0_DATA3	0x00000025
+				SC_P_EMMC0_DATA4_CONN_EMMC0_DATA4	0x00000025
+				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000025
+				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000025
+				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000025
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000045
+				SC_P_EMMC0_RESET_B_CONN_EMMC0_RESET_B	0x00000021
+			>;
+		};
+
+		pinctrl_usdhc1_200mhz: usdhc1grp200mhz {
+			fsl,pins = <
+				SC_P_EMMC0_CLK_CONN_EMMC0_CLK		0x06000047
+				SC_P_EMMC0_CMD_CONN_EMMC0_CMD		0x00000027
+				SC_P_EMMC0_DATA0_CONN_EMMC0_DATA0	0x00000027
+				SC_P_EMMC0_DATA1_CONN_EMMC0_DATA1	0x00000027
+				SC_P_EMMC0_DATA2_CONN_EMMC0_DATA2	0x00000027
+				SC_P_EMMC0_DATA3_CONN_EMMC0_DATA3	0x00000027
+				SC_P_EMMC0_DATA4_CONN_EMMC0_DATA4	0x00000027
+				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000027
+				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000027
+				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000027
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000047
 				SC_P_EMMC0_RESET_B_CONN_EMMC0_RESET_B	0x00000021
 			>;
 		};
 
 		pinctrl_usdhc2: usdhc2grp {
 			fsl,pins = <
-				SC_P_USDHC1_CLK_CONN_USDHC1_CLK		0x06000021
+				SC_P_USDHC1_CLK_CONN_USDHC1_CLK		0x06000041
 				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x00000021
 				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x00000021
 				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x00000021
@@ -229,8 +264,10 @@
 };
 
 &usdhc1 {
-	pinctrl-names = "default";
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	pinctrl-0 = <&pinctrl_usdhc1>;
+	pinctrl-1 = <&pinctrl_usdhc1_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc1_200mhz>;
 	bus-width = <8>;
 	non-removable;
 	status = "okay";
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index 9b519c9..079ec39 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -1379,8 +1379,11 @@
 			<&clk IMX8QM_SDHC0_CLK>,
 			<&clk IMX8QM_CLK_DUMMY>;
 		clock-names = "ipg", "per", "ahb";
-		assigned-clock-rates = <400000000>, <200000000>, <0>;
+		assigned-clocks = <&clk IMX8QM_SDHC0_DIV>;
+		assigned-clock-rates = <400000000>;
 		power-domains = <&pd_conn_sdch0>;
+		fsl,tuning-start-tap = <20>;
+		fsl,tuning-step= <2>;
 		status = "disabled";
 	};
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
index c0fb626..6c75149 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
@@ -107,7 +107,7 @@
 
 		pinctrl_usdhc1: usdhc1grp {
 			fsl,pins = <
-				SC_P_EMMC0_CLK_CONN_EMMC0_CLK		0x06000021
+				SC_P_EMMC0_CLK_CONN_EMMC0_CLK		0x06000041
 				SC_P_EMMC0_CMD_CONN_EMMC0_CMD		0x00000021
 				SC_P_EMMC0_DATA0_CONN_EMMC0_DATA0	0x00000021
 				SC_P_EMMC0_DATA1_CONN_EMMC0_DATA1	0x00000021
@@ -117,6 +117,41 @@
 				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000021
 				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000021
 				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000021
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000041
+				SC_P_EMMC0_RESET_B_CONN_EMMC0_RESET_B	0x00000021
+			>;
+		};
+
+		pinctrl_usdhc1_100mhz: usdhc1grp100mhz {
+			fsl,pins = <
+				SC_P_EMMC0_CLK_CONN_EMMC0_CLK		0x06000045
+				SC_P_EMMC0_CMD_CONN_EMMC0_CMD		0x00000025
+				SC_P_EMMC0_DATA0_CONN_EMMC0_DATA0	0x00000025
+				SC_P_EMMC0_DATA1_CONN_EMMC0_DATA1	0x00000025
+				SC_P_EMMC0_DATA2_CONN_EMMC0_DATA2	0x00000025
+				SC_P_EMMC0_DATA3_CONN_EMMC0_DATA3	0x00000025
+				SC_P_EMMC0_DATA4_CONN_EMMC0_DATA4	0x00000025
+				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000025
+				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000025
+				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000025
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000045
+				SC_P_EMMC0_RESET_B_CONN_EMMC0_RESET_B	0x00000021
+			>;
+		};
+
+		pinctrl_usdhc1_200mhz: usdhc1grp200mhz {
+			fsl,pins = <
+				SC_P_EMMC0_CLK_CONN_EMMC0_CLK		0x06000047
+				SC_P_EMMC0_CMD_CONN_EMMC0_CMD		0x00000027
+				SC_P_EMMC0_DATA0_CONN_EMMC0_DATA0	0x00000027
+				SC_P_EMMC0_DATA1_CONN_EMMC0_DATA1	0x00000027
+				SC_P_EMMC0_DATA2_CONN_EMMC0_DATA2	0x00000027
+				SC_P_EMMC0_DATA3_CONN_EMMC0_DATA3	0x00000027
+				SC_P_EMMC0_DATA4_CONN_EMMC0_DATA4	0x00000027
+				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000027
+				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000027
+				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000027
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000047
 				SC_P_EMMC0_RESET_B_CONN_EMMC0_RESET_B	0x00000021
 			>;
 		};
@@ -129,7 +164,7 @@
 
 		pinctrl_usdhc2: usdhc2grp {
 			fsl,pins = <
-				SC_P_USDHC1_CLK_CONN_USDHC1_CLK		0x06000021
+				SC_P_USDHC1_CLK_CONN_USDHC1_CLK		0x06000041
 				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x06000021
 				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x06000021
 				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x06000021
@@ -200,8 +235,10 @@
 };
 
 &usdhc1 {
-	pinctrl-names = "default";
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	pinctrl-0 = <&pinctrl_usdhc1>;
+	pinctrl-1 = <&pinctrl_usdhc1_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc1_200mhz>;
 	bus-width = <8>;
 	non-removable;
 	status = "okay";
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index c6a2f5a..401eb7f 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -913,8 +913,11 @@
 			<&clk IMX8QXP_SDHC0_CLK>,
 			<&clk IMX8QXP_CLK_DUMMY>;
 		clock-names = "ipg", "per", "ahb";
-		assigned-clock-rates = <400000000>, <200000000>, <0>;
+		assigned-clocks = <&clk IMX8QXP_SDHC0_DIV>;
+		assigned-clock-rates = <400000000>;
 		power-domains = <&pd_conn_sdch0>;
+		fsl,tuning-start-tap = <20>;
+		fsl,tuning-step= <2>;
 		status = "disabled";
 	};
 
-- 
1.7.9.5

